user_guide.adoc

```diff
@@ -1377,30 +1377,82 @@ library" for expression language.
 |Return type
 |Description
 
-|to_s
+|`to_s`
 |String
 |Converts integer into a string using decimal representation
-
-|to_s(radix)
-|String
-|Converts integer into a string using specified `radix` (i.e. use `16` to get hexadecimal representation, use `8` to get octal, etc)
 |===
 
 ==== Floating point numbers
 
-TODO
+No methods are currently supported for floating point numbers.
 
 ==== Byte arrays
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`to_s(encoding)`
+|String
+|Decodes (converts) a byte array encoded using the specified `encoding` scheme into a string
+|===
 
 ==== Strings
 
-TODO
+[cols="3*", options="header"]
+|===
+|Method name
+|Return type
+|Description
+
+|`length`
+|Integer
+|Length of a string in number of bytes
```

Comment:
"StrType" in general is supposed to represent strings of Unicode characters, to the extent supported natively by the language. That's `String` in Java, `string` in C#, `unicode` in Python, `String` with `encoding` set to `:UTF-8` in Ruby, etc. The whole point of such string types is that you can use it naturally, accessing characters, not bytes, without worrying about any encodings stuff, which is supposed to be accomplished on byte array -> string conversion step.

Unfortunately, situation with C++ is [generally so-so](http://doc.kaitai.io/lang_cpp_stl.html#_string_encoding). There's no universally accepted way to deal with Unicode characters — some people like to use various implementations of wide characters (`wchar_t`, `LPWSTR`, etc), some just store everything in UTF-8 and try to use UTF-8-aware operations, etc, etc.

Given that it wasn't a pressing issue for majority of formats, we've decided to just keep it simple for now and use basic C++ STL methods. Probably the way to go is to have several implementations, selectable using some sort of `#define` in compile-time (i.e. iconv-based implementation with "convert everything to UTF-8 internally" principle, Windows-based implementation using WinAPI stuff, ICU-based, etc, etc).

Given that you use C++, your input here is more than welcome — I'm totally not a C++ expert here :)